Samplebuilder: add tests for padding packets

tmatth opened the pull request:
Refs #2323

Co-authored-by: Juliusz Chroboczek <jch@irif.fr>

k0nserv commented on pkg/media/samplebuilder/samplebuilder_test.go:
```diff
@@ -223,6 +223,27 @@ func TestSampleBuilder(t *testing.T) {
+				{Header: rtp.Header{SequenceNumber: 5002, Timestamp: 1, Marker: true}, Payload: []byte{3}},
+				{Header: rtp.Header{SequenceNumber: 5003, Timestamp: 1}, Payload: []byte{}},
+				{Header: rtp.Header{SequenceNumber: 5004, Timestamp: 1}, Payload: []byte{}},
+				{Header: rtp.Header{SequenceNumber: 5005, Timestamp: 3}, Payload: []byte{1}},
```
It's worth having a test that also has `Marker: true` here, i.e. a packet that follows a run of padding packets and is both a start and end of a partition i.e. a very small video partition that fits inside 1200 bytes. This is something I observed with Chrome and it's covered in the webrtc-rs code.

tmatth replied:
@k0nserv would that be this one ?
https://github.com/webrtc-rs/webrtc/pull/309/files#diff-0e05f283d453fbc18b4b99c7bf67c1ccd8e630a44e57047ee7a2a5fcc70dfbccR1055

k0nserv replied:
That's testing that it can built a sample from such a packet in isolation. This one test such a packet following a padding run https://github.com/webrtc-rs/webrtc/pull/309/files#diff-0e05f283d453fbc18b4b99c7bf67c1ccd8e630a44e57047ee7a2a5fcc70dfbccR1099 where "Valid packet 3"  is both start and end of a partition

tmatth replied:
I've added a test for this with some caveats:
- for some reason setting `withHeadChecker: true` crashes on trying to index 0 of an empty array, I'm guessing it's due to missing padding handling?
- likewise there is no `PrevPaddingPackets` field yet
- and the expected results are not the same as in webrtc-rs, presumably again because padding isn't yet handled here

Sean-Der commented:
@tmatth Mind looking at this one more time before I merge?

I rebased and removed the new public API members. No one has requested them, so don't want to expand the public API unless needed!

Sorry this took so long :( just working on getting PRs down to zero now.

tmatth commented:
> @tmatth Mind looking at this one more time before I merge?
> 
> I rebased and removed the new public API members. No one has requested them, so don't want to expand the public API unless needed!
> 
> Sorry this took so long :( just working on getting PRs down to zero now.

Thanks! LGTM, I still never really got to the bottom of the main issue I wanted to address whereby A/V sync would diverge if there was a long interruption in video packets (not lost, just nothing being sent by the publisher for a few seconds). Any thoughts?

Sean-Der commented:
@tmatth i think you will need to delay using the sender reports? I can help implement that

tmatth commented:
> @tmatth i think you will need to delay using the sender reports? I can help implement that

Can this be done internally in `SampleBuilder` or is it up to users of `SampleBuilder` to handle that?
